src/workerd/io/worker-entrypoint.c++

```diff
@@ -551,11 +551,18 @@ kj::Promise<WorkerInterface::AlarmResult> WorkerEntrypoint::runAlarmImpl(
 
       try {
         auto result = co_await context.run(
-            [scheduledTime, entrypointName=entrypointName, &context](Worker::Lock& lock){
+            [scheduledTime, timeout=context.getLimitEnforcer().getAlarmLimitMs(), entrypointName=entrypointName, &context](Worker::Lock& lock){
           jsg::AsyncContextFrame::StorageScope traceScope = context.makeAsyncTraceScope(lock);
 
+          // If we have an invalid timeout, set it to the default value of 15 minutes.
+          auto timeoutMs = timeout;
+          if (timeout == 0 * kj::MILLISECONDS) {
+            LOG_NOSENTRY(WARNING, "Invalid alarm timeout value. Using 900000ms", timeout);
+            timeoutMs = 900000 * kj::MILLISECONDS;
```

Comment:
What's nice about using a duration is that you can set it to `15 * kj::MINUTES`, which is a whole lot easier to read.  (You may want to update the log message too.)